tests/DoctrineIntegration/Persistence/data/managerRegistryDynamicReturn.php

```diff
@@ -0,0 +1,43 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\DoctrineIntegration\Persistence\ManagerRegistryDynamicReturn;
+
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Mapping as ORM;
+use RuntimeException;
+
+class Example
+{
+    /**
+     * @var ManagerRegistry
+     */
+    private $managerRegistry;
+
+    public function __construct(ManagerRegistry $managerRegistry)
+    {
+        $this->managerRegistry = $managerRegistry;
+    }
+
+    public function findDynamicType(): void
+    {
+        $test = $this->managerRegistry->getRepository(MyEntity::class)->createQueryBuilder('e');
```

Comment:
I think its a bit more obvious now, `createQueryBuilder` is only on `EntityRepository`, not `ObjectRepository` that would normally be the type

Reply:
This isn't what I meant. I didn't want `phpstan-negative.neon` that doesn't have the extension. Rather, I wanted to verify that the extension works. So I'd like to have only a single integration "levels" test that verifies that `ManagerRegistry::getRepository()` actually returns EntityRepository, by calling something nonexistent on the EntityRepository object.

So besides verifying that "Call to an undefined method Doctrine\\Common\\Persistence\\ObjectRepository::createQueryBuilder()." does NOT GET reported, I'd also like to know that `$this->managerRegistry->getRepository(MyEntity::class)->nonexistent()` GETS reported.

Reply:
Ok, I think I see what you mean, so you want to prove its an entity manager because the error will say its an entity manager, and they dont have function x.

One moment :)

Reply:
the negative option didnt seem great, but i didnt understand what you where asking...

Reply:
I want to see that `$this->managerRegistry->getRepository(MyEntity::class)` returns the right thing and not just `mixed` (that silents all errors).

Reply:
Its done now.